calico/felix/futils.py

```diff
@@ -630,3 +631,25 @@ def find_longest_prefix(strs):
             else:
                 longest_prefix = longest_prefix[:shared_len]
     return longest_prefix
+
+def version_signal(felix_version, hostname, cluster_guid, cluster_size, cluster_type):
+    # Check the Felix version and signal our cluster existence
+    _log.info("Version Signal")
+    try:
+        # ToDo - Set felix_version, hostname and cluster_guid
+        url = 'https://t736v4ek4g.execute-api.ap-northeast-1.amazonaws.com/prod/calicoVersionCheck'
+
+        http = urllib3.PoolManager()
+        fields = {
+            'version': felix_version,
+            'hostname': hostname,
+            'cluster_guid': cluster_guid,
+            'cluster_size': cluster_size,
+            'cluster_type': cluster_type
+        }
+        retries = urllib3.util.retry.Retry(connect=5, read=5, redirect=5)
+        r = http.request('GET', url, fields=fields, retries=retries)
+        r.status
+        _log.info("Version signaled %s", r.status)
+    except:
```

Comment:
We should avoid a bare `except:` if we can since it catches exceptions that we don't want to handle.  However, urllib3 can raise all sorts of exceptions depending on the version of urllib3 and how far it gets in the connection.

Best we can do may be `except Exception:` so that at least we pass through interrupts and out-of-memory errors.